File: todoDesktop/socket_client.py
import socket
import json

class SocketClient:
    def __init__(self, host="127.0.0.1", port=8080):
        self.host = host
        self.port = port
        self.sock = None

    def send_and_receive(self, data: dict) -> dict:
        try:
            with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as sock:
                sock.connect((self.host, self.port))  
                sock.sendall(json.dumps(data).encode())
                response_data = sock.recv(4096)
                return json.loads(response_data.decode())
        except Exception as e:
            print(f"Erro de socket: {e}")
            return {"success": False, "error": str(e)}
    
    def connect(self):
        self.sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
        self.sock.connect((self.host, self.port))

    def close(self):
        if self.sock:
            self.sock.close()
            self.sock = None

File: src/services/SocketClient.ts
export class SocketClient {
  private host: string;
  private port: number;

  constructor(host: string = '192.168.0.29', port: number = 5000) {
    this.host = host;
    this.port = port;
  }

  async sendAndReceive(data: any): Promise<any> {
    try {
      const response = await fetch(`http://${this.host}:${this.port}`, {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify(data),
      });

      if (!response.ok) {
        throw new Error('Network response was not ok');
      }

      return await response.json();
    } catch (error: any) {
      console.error('Socket error:', error);
      return { success: false, error: error.message || 'Erro de conexão' };
    }
  }
}